amd/src/courseformat/courseindex/placeholder.js

```diff
@@ -69,4 +70,48 @@ export default class Component extends BaseComponent {
             throw error;
         }
     }
+
+     _nestSections(sections) {
+        let topSections = [];
+        let parentSection = {};
+        let lastParent = {};
+
+        // Let's re-organise our sections.
+        for (let i = 0; i < sections.length; i++) {
+            let section = sections[i];
+            let isCurrent = false;
+            section.indexcollapsed = true;
+            section.contentcollapsed = true;
+            section.subsections = [];
+            section.topics = [];
+            if (isCurrent) {
+                section.indexcollapsed = false;
+                section.contentcollapsed = false;
+            }
+            if (window.location.href == section.sectionurl.replace(/&amp;/g, "&")) {
+                isCurrent = true;
+            }
+            if (section.indent === 0) {
+                parentSection = section;
+                lastParent = section;
+                topSections.push(section);
+            } else if (section.indent === 1) {
+                lastParent = section;
+                parentSection.subsections.push(section);
+                if (isCurrent) {
+                    parentSection.indexcollapsed = false;
+                    parentSection.contentcollapsed = false;
+                }
+            } else if (section.indent === 2) {
+                lastParent.topics.push(section);
+                if (isCurrent) {
+                    parentSection.indexcollapsed = false;
+                    parentSection.contentcollapsed = false;
+                    lastParent.indexcollapsed = false;
+                    lastParent.contentcollapsed = false;
+                }
+            }
+        }
```

Comment:
Setting collapse state here seems to cause issues for the drag and drop code, this could instead be set in classes/output/courseformat/content.php (code for this pasted in the main comment).  A downside is it won't have access to the anchor part of the URL, so it can't expand the topic within the page, but I think it's good enough.
```suggestion
        // Let's re-organise our sections.
        for (let i = 0; i < sections.length; i++) {
            let section = sections[i];
            section.subsections = [];
            section.topics = [];
            if (section.indent === 0) {
                parentSection = section;
                lastParent = section;
                topSections.push(section);
            } else if (section.indent === 1) {
                lastParent = section;
                parentSection.subsections.push(section);
            } else if (section.indent === 2) {
                lastParent.topics.push(section);
            }
        }
```